crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -92,6 +95,7 @@ pub(crate) struct EnumVariant<'a> {
     /// The fields within this variant.
     pub fields: EnumVariantFields<'a>,
     /// The reflection-based attributes on the variant.
+    #[allow(dead_code)]
```

Comment:
This shouldn't be needed as the field is pub.

Reply:
I don't believe this is the case, since we're in the context of a proc_macro crate, everything is visible at the level of `pub(crate)` meaning we need these. With removal I get:
```rust
error: fields `attrs` and `index` are never read
   --> crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs:98:5
    |
92  | pub(crate) struct EnumVariant<'a> {
    |                   ----------- fields in this struct
...
98  |     pub attrs: ReflectFieldAttr,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
99  |     /// The index of this variant within the enum.
100 |     pub index: usize,
    |     ^^^^^^^^^^^^^^^^
    |
    = note: `-D dead-code` implied by `-D warnings`

error: associated function `serialization_denylist` is never used
   --> crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs:[27](https://github.com/bevyengine/bevy/actions/runs/3081629201/jobs/4980397401#step:6:28)1:12
    |
271 |     pub fn serialization_denylist(&self) -> &BitSet<u32> {
    |            ^^^^^^^^^^^^^^^^^^^^^^

error: associated function `fields` is never used
   --> crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs:314:12
    |
314 |     pub fn fields(&self) -> &[StructField<'a>] {
    |            ^^^^^^
```